internal/blockchain/v0/pool.go

```diff
@@ -216,6 +223,14 @@ func (pool *BlockPool) PopRequest() {
 		delete(pool.requesters, pool.height)
 		pool.height++
 		pool.lastAdvance = time.Now()
+
+		// the lastSyncRate will be updated every 100 blocks, it uses the adaptive filter
+		// to smooth the block sync rate and the unit represents the number of blocks per second.
+		if pool.height%100 == 0 {
+			pool.lastSyncRate = 0.9*pool.lastSyncRate + 0.1*(100/time.Since(pool.lastHundredBlockTimeStamp).Seconds())
```

Comment:
If this is the first iteration of the loop, pool.lastSyncRate will be 0 which will make this number 1/10th of the actual rate and take a while to actually reach something close to the true value. I think we should add a clause for the first iteration to set the rate at:
```
100/time.Since(pool.lastHundredBlockTimeStamp).Seconds()
```

Reply:
@cmwaters 
It was the algo from v2 blockchain reactor, do you think to change it too?

My thinking was it is just an estimation, so it doesn't really matter if the node only has few hundred blocks to sync.
same as `pool.height%100`

Reply:
Yeah I think it's worth checking if `lastSyncRate` is 0 to set it to `100/time.Since(pool.lastHundredBlockTimeStamp).Seconds()`
Perhaps it's fine to not worry about the `pool.height%100`